### DISCUSSION TITLE
kvrocks running on kubernetes

### DISCUSSION THREAD

**Discussion by linydquantil** [apache/kvrocks#359] [Q&A] [answered]
Have you tried to run to K8s?
Is there any experience to share?

**Comment by git-hulk** [comment]
@smartlee Ctrip's kvrocks instances were running on the Kubernetes, maybe we can transfer this issue to the discussion.

**Comment by smartlee** [comment, marked as answer]
@linydquantil  yes,our kvrocks clusters running on  k8s about  1 years+ , and Redis on k8s over 3 years, u can  refer to those articles :
https://mp.weixin.qq.com/s/higbRyeGtNLMEAqtUG6ZVg
https://mp.weixin.qq.com/s/L7EQzthCoWuJw8TzY-KAMw
https://mp.weixin.qq.com/s/QTqcBZlAhp5cLRJGJVZRNw

**Reply by anupriya17** [reply]
The links are not working. Can you share the kvrocks.conf and kubernetes yaml files if possible? Thank you

**Reply by smartlee** [reply]
why? kvrock.conf is just the same as the offical one in this repo with a little difference. and yaml is customized and has some scheduling rules, these are not suitable for you

**Reply by anupriya17** [reply]
Thanks for response. In redis cluster configuration is easy it uses gossip protocol. But kvrocks has no such setting. So I don't know how to configure in yamls.

**Reply by smartlee** [reply]
cluster mode in our company is consistent hash, so just set up some shards, every shard is StatefulSet with some replicas.https://mp.weixin.qq.com/s/higbRyeGtNLMEAqtUG6ZVg this article describes details about how we maintain large-scale Redis cluters in our company, exactly the same for kvrocks

**Comment by linydquantil** [comment]
@smartlee ok, Thank you for sharing.
